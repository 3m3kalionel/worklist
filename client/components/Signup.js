import React from 'react';
import { Mutation } from 'react-apollo';
import Link from 'next/link';

import AuthForm from '../styles/AuthForm';
import { WithForm } from './HOC';
import SIGN_UP_MUTATION from '../graphql/mutations/Signup';

const Signup = props => {
  const { onChange, onSubmit, onerror, values } = props;
  const { username, email, password, phoneNumber } = values;

  return (
    <Mutation
      mutation={SIGN_UP_MUTATION}
      variables={values}
    >
      {(signup, { error, loading }) => (
        <form
          onSubmit={onSubmit(signup, 'signup')}
        >
          <AuthForm>
            <div id="authentication-form">
              <div id="form-body">
                <h1 id="form-header">Sign Up For Free</h1>
                <label htmlFor="username">Username</label>
                <input
                  name="username"
                  type="text"
                  className="auth-form-input"
                  placeholder="Username"
                  value={username}
                  autoComplete="username"
                  onChange={onChange}
                  required
                />
                <label htmlFor="phoneNumber">Telephone</label>
                <input
                  name="phoneNumber"
                  type="number"
                  className="auth-form-input"
                  placeholder="Phone Number"
                  value={phoneNumber}
                  autoComplete="tel"
                  onChange={onChange}
                  required
                />
                <label htmlFor="email">Email</label>
                <input
                  name="email"
                  type="email"
                  className="auth-form-input"
                  placeholder="Email"
                  value={email}
                  autoComplete="email"
                  onChange={onChange}
                  required
                />
                <label htmlFor="password">Password</label>
                <input
                  name="password"
                  type="password"
                  className="auth-form-input"
                  placeholder="Password"
                  value={password}
                  autoComplete="current-password"
                  onChange={onChange}
                  required
                />
                <div id="action-button-container">
                  <button
                    type="submit"
                    id="sign-up-button"
                    disabled={loading}
                  >
                    Sign Up
                  </button>
                  <span
                    id="forgot-password"
                  >
                    <Link href="/requestreset">
                      <a>Forgot Password?</a>
                    </Link>
                  </span>
                </div>
              </div>
            </div>
          </AuthForm>
        </form>
      )}
    </Mutation>
  )
}

export default WithForm(Signup);
