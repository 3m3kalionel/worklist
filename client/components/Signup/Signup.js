import React from 'react';
import { Mutation } from 'react-apollo';
import Link from 'next/link';

import AuthFormStyles from '../../styles/AuthFormStyles';
import WithForm from '../HOC';

import SIGN_UP_MUTATION from '../../graphql/mutations/Signup';

import CURRENT_USER_QUERY from '../../graphql/queries/CurrentUser';

const Signup = props => {
  const { onChange, onSubmit, onError, values } = props;
  const { username, email, password, phoneNumber } = values;

  return (
    <Mutation
      mutation={SIGN_UP_MUTATION}
      variables={values}
      refetchQueries={[{ query: CURRENT_USER_QUERY }]}
    >
      {(signup, { error, loading }) => (
        <form
          data-testid="form"     
          onSubmit={onSubmit(signup, 'signup')}
        >
          <AuthFormStyles>
            <div id="authentication-form">
              <div id="form-body">
                {error && onError(error)}
                {props.children}
                <h1 id="form-header">Sign Up For Free</h1>
                <label htmlFor="username">Username</label>
                <input
                  name="username"
                  type="text"
                  className="auth-form-input"
                  placeholder="Username"
                  value={username}
                  autoComplete="username"
                  onChange={onChange}
                  required
                />
                <label htmlFor="phoneNumber">Telephone</label>
                <input
                  name="phoneNumber"
                  type="number"
                  className="auth-form-input"
                  placeholder="Phone Number"
                  value={phoneNumber}
                  autoComplete="tel"
                  onChange={onChange}
                  required
                />
                <label htmlFor="email">Email</label>
                <input
                  name="email"
                  type="email"
                  className="auth-form-input"
                  placeholder="Email"
                  value={email}
                  autoComplete="email"
                  onChange={onChange}
                  required
                />
                <label htmlFor="password">Password</label>
                <input
                  name="password"
                  type="password"
                  className="auth-form-input"
                  placeholder="Password"
                  value={password}
                  autoComplete="current-password"
                  onChange={onChange}
                  required
                />
                <div id="action-button-container">
                  <button
                    type="submit"
                    id="sign-up-button"
                    disabled={loading}
                  >
                    Sign Up
                  </button>
                  <span
                    id="forgot-password"
                  >
                    <Link href="/forgotpassword">
                      <a>Forgot Password?</a>
                    </Link>
                  </span>
                </div>
              </div>
            </div>
          </AuthFormStyles>
        </form>
      )}
    </Mutation>
  )
}

export default WithForm(Signup);
