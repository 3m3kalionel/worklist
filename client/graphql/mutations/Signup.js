import gql from 'graphql-tag';

const SIGN_UP_MUTATION = gql`
  mutation SIGN_UP_MUTATION(
    $username: String!,
    $email: String!,
    $phoneNumber: String!,
    $password: String!,
  ) {
    signup(
      username: $username,
      email: $email,
      phoneNumber: $phoneNumber,
      password: $password
    ) {
      id
      email
      username
    }
  }
`;

export default SIGN_UP_MUTATION;
