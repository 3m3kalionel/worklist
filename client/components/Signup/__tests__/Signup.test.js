import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ApolloProvider } from 'react-apollo';
import '@testing-library/react/cleanup-after-each';
import { createMockClient } from 'mock-apollo-client';

import Signup from '../Signup';

import SIGN_UP_MUTATION from '../../../graphql/mutations/RequestReset';

jest.mock("../../HOC.js", () => Component => {
  const mockHOC = props => (
    <div><Component {...props} /></div>
  );

   return mockHOC;
});

const setup = (values, onSubmit) => {
  const onSubmitHandler = onSubmit || jest.fn();
  const mockClient = createMockClient();

  mockClient.setRequestHandler(
    SIGN_UP_MUTATION,
    () => Promise.resolve({
      data: {
        signup: {
          id: "cjzdoavvjvgby0b53mie0fqc3",
          username: "gee",
          email: "gee@gmail.com",
          profilePicUrl: null,
          authoredTodoLists: [],
          todoListsContributedTo: []
        }
      }
    })
  );

  const {
    container,
    getByText,
    getByTestId
  } = render(
    <ApolloProvider client={mockClient}>
      <Signup
        values={values}
        onSubmit={onSubmitHandler}
        onChange={jest.fn()}
      />
    </ApolloProvider>
  );

  return { container, getByText, getByTestId }
}

describe('<Signup />', () => {
  test('renders an empty form', async () => {
    const { container } = setup({
      usernname: "",
      email: "",
      password: "",
      phoneNumber: ""
    });

    expect(container).toMatchSnapshot()
  });

  test('renders text in the form input', () => {
    const { container } = setup({
      usernname: "username",
      email: "email@gmail.com",
      password: "password",
      phoneNumber: "phoneNumber"
    });

    expect(container).toMatchSnapshot();
  });

  test('calls the submit handler when button is clicked', () => {
  const onSubmitHandler = jest.fn();

    const {
      container,
      getByTestId
    } = setup({
      usernname: "username",
      email: "email@gmail.com",
      password: "password",
      phoneNumber: "phoneNumber"
      },
      onSubmitHandler
    );
    const form = getByTestId('form');
    fireEvent.submit(form);

    expect(onSubmitHandler).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
