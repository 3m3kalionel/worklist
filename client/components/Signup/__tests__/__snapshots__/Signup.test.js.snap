// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> calls the submit handler when button is clicked 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div
          id="authentication-form"
        >
          <div
            id="form-body"
          >
            <h1
              id="form-header"
            >
              Sign Up For Free
            </h1>
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="username"
              class="auth-form-input"
              name="username"
              placeholder="Username"
              required=""
              type="text"
              value=""
            />
            <label
              for="phoneNumber"
            >
              Telephone
            </label>
            <input
              autocomplete="tel"
              class="auth-form-input"
              name="phoneNumber"
              placeholder="Phone Number"
              required=""
              type="number"
              value="phoneNumber"
            />
            <label
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="email"
              class="auth-form-input"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value="email@gmail.com"
            />
            <label
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="auth-form-input"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value="password"
            />
            <div
              id="action-button-container"
            >
              <button
                id="sign-up-button"
                type="submit"
              >
                Sign Up
              </button>
              <span
                id="forgot-password"
              >
                <a
                  href="/forgotpassword"
                >
                  Forgot Password?
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Signup /> renders an empty form 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div
          id="authentication-form"
        >
          <div
            id="form-body"
          >
            <h1
              id="form-header"
            >
              Sign Up For Free
            </h1>
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="username"
              class="auth-form-input"
              name="username"
              placeholder="Username"
              required=""
              type="text"
              value=""
            />
            <label
              for="phoneNumber"
            >
              Telephone
            </label>
            <input
              autocomplete="tel"
              class="auth-form-input"
              name="phoneNumber"
              placeholder="Phone Number"
              required=""
              type="number"
              value=""
            />
            <label
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="email"
              class="auth-form-input"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
            <label
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="auth-form-input"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
            <div
              id="action-button-container"
            >
              <button
                id="sign-up-button"
                type="submit"
              >
                Sign Up
              </button>
              <span
                id="forgot-password"
              >
                <a
                  href="/forgotpassword"
                >
                  Forgot Password?
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Signup /> renders text in the form input 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div
          id="authentication-form"
        >
          <div
            id="form-body"
          >
            <h1
              id="form-header"
            >
              Sign Up For Free
            </h1>
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="username"
              class="auth-form-input"
              name="username"
              placeholder="Username"
              required=""
              type="text"
              value=""
            />
            <label
              for="phoneNumber"
            >
              Telephone
            </label>
            <input
              autocomplete="tel"
              class="auth-form-input"
              name="phoneNumber"
              placeholder="Phone Number"
              required=""
              type="number"
              value="phoneNumber"
            />
            <label
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="email"
              class="auth-form-input"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value="email@gmail.com"
            />
            <label
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="auth-form-input"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value="password"
            />
            <div
              id="action-button-container"
            >
              <button
                id="sign-up-button"
                type="submit"
              >
                Sign Up
              </button>
              <span
                id="forgot-password"
              >
                <a
                  href="/forgotpassword"
                >
                  Forgot Password?
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
